tools/build_defs/framework.bzl

```diff
@@ -309,6 +313,33 @@ def cc_external_rule_impl(ctx, attrs):
         ),
     ]
 
+def _define_ld_library_path(deps):
+    """Return a list of library paths for all transitive dependencies.
+
+    Linux can have trouble resolving libraries that aren't in standard paths
+    [0,1,2].  Some [1] configure scripts actually compile test programs to test
+    system configuration and on Linux, without LD_LIBRARY_PATH set these
+    configure scripts fail.  I (bshi) haven't had issues with this on MacOS and
+    purportedly this is only an issue on Linux so ideally we'd constrain to
+    Linux but see [3].
+
+    [0] https://github.com/bazelbuild/rules_foreign_cc/issues/241
+    [1] https://www.postgresql.org/message-id/23645.1437968936%40sss.pgh.pa.us
+    [2] https://github.com/bazelbuild/rules_foreign_cc/pull/247
+    [3] https://github.com/bazelbuild/bazel/issues/11107.
```

Comment:
This seems like something that's now possible

Reply:
Yes but linux isn't the only platform that uses `LD_LIBRARY_PATH` e.g. freeBSD so I don't think we should put this comment in as is as its not accurate. On macOS the env var is `DYLD_LIBRARY_PATH` but I don't fully understand when this is required.

Reply:
I've updated the language of the comment to clarify that this is a known issue on but potentially not restricted to Linux.